engine/runtime/src/main/scala/org/enso/interpreter/instrument/job/ProgramExecutionSupport.scala

```diff
@@ -351,27 +351,32 @@ object ProgramExecutionSupport {
               VisualizationResult.findExceptionMessage(panic),
               ErrorResolver.getStackTrace(panic).flatMap(_.expressionId)
             )
-        case withWarnings: WithWarnings =>
-          val warningsCount = withWarnings.getWarningsCount
-          val warning =
-            if (warningsCount == 1) {
-              val warnings = withWarnings.getWarningsArray(null)
-              Option(ctx.executionService.toDisplayString(warnings(0).getValue))
-            } else {
-              None
-            }
-          Api.ExpressionUpdate.Payload.Value(
-            Some(
-              Api.ExpressionUpdate.Payload.Value
-                .Warnings(
-                  warningsCount,
-                  warning,
-                  withWarnings.isLimitReached()
+        case _ =>
+          if (WarningsLibrary.getUncached().hasWarnings(value.getValue)) {
+            val warnings =
+              WarningsLibrary.getUncached().getWarnings(value.getValue, null)
+            val warningsCount = warnings.length
+            val warning =
+              if (warningsCount >= 1) {
```

Comment:
This changes the semantics. We only want to include the warning if it is a single one.

Reply:
OK. Semantics is back in ba86adf - however I have to admit, I don't understand the reasoning - why do we want a single warning?